storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java

```diff
@@ -162,6 +185,191 @@ public void testGetUserId() throws Exception {
+            if (i < pidList.size() - 1) {
+                if (pidDirsAvailable) {
+                    LOG.info("{}: ({}) Found existing process directories before killing last process", testName, i);
+                } else {
+                    String e = String.format("%s: (%d) Found no existing process directories before killing last process", testName, i);
+                    errors.add(e);
+                    LOG.error(e);
+                }
+            } else {
+                if (pidDirsAvailable) {
+                    String e = String.format("%s: (%d) Found existing process directories after killing last process", testName, i);
+                    errors.add(e);
+                    LOG.error(e);
+                } else {
+                    LOG.info("{}: ({}) Found no existing process directories after killing last process", testName, i);
+                }
+            }
+        }
+        if (!errors.isEmpty()) {
+            fail(String.format("There are %d failures in test:\n\t%s", errors.size(), String.join("\n\t", errors)));
+        }
+    }
+
+
+    /**
+     * Simulate the production scenario where the owner of the process directory is sometimes returned as the
+     * UID instead of user. This scenario is simulated by calling
+     * {@link ServerUtils#isAnyPosixProcessPidDirAlive(Collection, String, boolean)} with the last parameter
+     * set to true as well as false.
+     *
+     * @throws Exception on I/O exception
+     */
+    @Test
+    public void testIsAnyPosixProcessPidDirAliveMockingFileOwnerUid() throws Exception {
+        File procDir = new File("/proc");
+        if (!procDir.exists()) {
+            LOG.info("Test testIsAnyPosixProcessPidDirAlive is designed to run on systems with /proc directory only, marking as success");
+            return;
+        }
+        Collection<Long> pids = getRunningProcessIds();
+        assertFalse(pids.isEmpty());
+
+        for (int i = 0 ; i < 2 ; i++) {
+            boolean mockFileOwnerToUid = (i % 2 == 0);
+            // at least one pid will be owned by the current user (doing the testing)
+            String currentUser = System.getProperty("user.name");
+            boolean status = ServerUtils.isAnyPosixProcessPidDirAlive(pids, currentUser, mockFileOwnerToUid);
+            String err = String.format("(mockFileOwnerToUid=%s) Expecting user %s to own at least one process",
+                    mockFileOwnerToUid, currentUser);
+            assertTrue(err, status);
+        }
+    }
+
+    /**
+     * Make the best effort to obtain the Process ID from the Process object. Thus staying entirely with the JVM.
+     *
+     * @param p Process instance returned upon executing {@link Runtime#exec(String)}.
+     * @param errors Populate errors when PID is a negative number.
+     * @return positive PID upon success, otherwise negative.
+     */
+    private synchronized long getPidOfPosixProcess(Process p, List<String> errors) {
```

Comment:
this method already exist?

Reply:
I am not aware of any existing method with similar functionality.

Reply:
https://github.com/apache/storm/blob/master/storm-server/src/test/java/org/apache/storm/utils/ServerUtilsTest.java#L266
This method already exists in ServerUtilsTest (from your another merged PR)